random-name/src/App.js

```diff
@@ -2,31 +2,59 @@ import React, {useState, useEffect} from 'react';
 import './App.css';
 
 
-const students = ['joe ', 'bob ', 'billy ', 'isaac '];
+const students = [
+  {
+    id: 1,
+    class:'math',
+    studentNames: ['joe ', 'bob ', 'billy ', 'isaac '],
+  },
+  {
+    id: 2,
+    class: 'science',
+    studentNames: ['fred', 'chris', 'zack', 'shawn', 'joshua', 'amy'],
+  }
+];
 
 function App() {
   const [names, setNames] = useState(students);
-  const [randomName, setRandomName] = useState('')
+  const [randomName, setRandomName] = useState('');
+  const [selectedClass, setSelectedClass] = useState(1);
   const min = 0;
-  const max = names.length;
+  const max = names[Number(selectedClass - 1)].studentNames.length;
```

Comment:
This line declares a variable max and assigns an integer = the length of studentNames list in a selectedClass.